internal/pkg/db/mongo/metadata.go

```diff
@@ -1037,7 +1036,14 @@ func (m *MongoClient) UpdateCommand(c *contract.Command, r *contract.Command) er
 		r.Origin = c.Origin
 	}
 
-	return col.UpdateId(id, r)
+	var err error
+	if model.Id.Valid() {
+		err = s.DB(m.database.Name).C(db.Command).UpdateId(model.Id, model)
+	} else {
+		query := bson.M{"uuid": model.Uuid}
+		err = s.DB(m.database.Name).C(db.Command).Update(query, model)
+	}
```

Comment:
Help me understand -- We instantiate a models.Command above (line 1018) and populate it from the contract. We use that instance in the UpdateId/Update calls, but we never set any new values on it. Since we map it from `c`, inferring only from the logic above, I guess the `model` instance contains all the new values we need, so we just update. If that's so, why preserve the lines setting `r` properties from `c`? It looks to me like `r` is useless here.

Reply:
Inherited code but will take responsibility for it.  Refactored to remove second parameter from dbClient.UpdateCommand().